src/core/dbus-service.c

```diff
@@ -532,6 +564,32 @@ static int bus_service_set_transient_property(
         if (streq(name, "StandardErrorFileDescriptor"))
                 return bus_set_transient_std_fd(u, name, &s->stderr_fd, &s->exec_context.stdio_as_fds, message, flags, error);
 
+        if (streq(name, "OpenFile")) {
+                const char *open_file_str;
+
+                r = sd_bus_message_enter_container(message, 'a', "s");
```

Comment:
Same here, we should read structured data instead of a string over D-Bus. So we should also be parsing an array of D-Bus structs here, path, fd name, and bitflags stored as a 64-bit integer for the options.